From b935f1ce1b13199f3327da86e5d737e55f8c2513 Mon Sep 17 00:00:00 2001
From: Wen Gong <quic_wgong@quicinc.com>
Date: Tue, 18 Oct 2022 04:04:35 -0400
Subject: [PATCH 109/112] wifi: mac80211: temp remove link_info_changed
 callback for single mlo link

Temp workaround to avoid crash if sdata->vif.bss_conf
link id is 0, wherease the actual value is 2.

Signed-off-by: Wen Gong <quic_wgong@quicinc.com>
Signed-off-by: Amutha Ravikumar <quic_aravikum@quicinc.com>
---
 net/mac80211/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/net/mac80211/main.c b/net/mac80211/main.c
index 5c7a97921cf0..026718b276fa 100644
--- a/net/mac80211/main.c
+++ b/net/mac80211/main.c
@@ -243,7 +243,8 @@ void ieee80211_bss_info_change_notify(struct ieee80211_sub_if_data *sdata,
 			local->ops->vif_cfg_changed(&local->hw, &sdata->vif, ch);
 	}
 
-	if (changed & ~BSS_CHANGED_VIF_CFG_FLAGS) {
+	if (((!sdata->vif.valid_links) || (sdata->vif.valid_links & BIT(0))) &&
+	    changed & ~BSS_CHANGED_VIF_CFG_FLAGS) {
 		u64 ch = changed & ~BSS_CHANGED_VIF_CFG_FLAGS;
 
 		/* FIXME: should be for each link */
-- 
2.31.1

